includes/class-webmention-mf2-handler.php

```diff
@@ -0,0 +1,389 @@
+<?php
+/**
+ * takes MF2 JSON and processes it
+ *
+ */
+class Webmention_MF2_Handler {
+
+
+	/**
+	 * Is string a URL.
+	 *
+	 * @param array $string
+	 * @return bool
+	*/
+	public static function is_url( $string ) {
+		if ( ! is_string( $string ) ) {
+			return false;
+		}
+		// If debugging is on just validate that URL is validly formatted
+		if ( WP_DEBUG ) {
+			return filter_var( $string, FILTER_VALIDATE_URL ) !== false;
+		}
+		// If debugging is off limit based on WordPress parameters
+		return wp_http_validate_url( $string );
+	}
+
+	/**
+	 * is this what type
+	 *
+	 * @param array $mf Parsed Microformats Array
+	 * @param string $type Type
+	 * @return bool
+	*/
+	public static function is_type( $mf, $type ) {
+		return is_array( $mf ) && ! empty( $mf['type'] ) && is_array( $mf['type'] ) && in_array( $type, $mf['type'], true );
+	}
+
+	/**
+	 * Verifies if $mf is an array without numeric keys, and has a 'properties' key.
+	 *
+	 * @param $mf
+	 * @return bool
+	 */
+	public static function is_microformat( $mf ) {
+		return ( is_array( $mf ) && ! wp_is_numeric_array( $mf ) && ! empty( $mf['type'] ) && isset( $mf['properties'] ) );
+	}
+
+	/**
+	 * Verifies if $mf has an 'items' key which is also an array, returns true.
+	 *
+	 * @param $mf
+	 * @return bool
+	 */
+	public static function is_microformat_collection( $mf ) {
+		return ( is_array( $mf ) && isset( $mf['items'] ) && is_array( $mf['items'] ) );
+	}
+
+	/**
+	 * Verifies if property named $propname is in array $mf.
+	 *
+	 * @param array    $mf
+	 * @param $propname
+	 * @return bool
+	 */
+	public static function has_prop( array $mf, $propname ) {
```

Comment:
Most of the functions have full/representative names, so I would prefer `has_property`.

Reply:
That's fine. Some of these functions are based on the old mf-cleaner repo, so the names are inherited from that. Will adapt.